gateway/src/traits.rs

```diff
@@ -42,7 +52,111 @@ impl WeightInfo for () {
     fn approve_contract_call() -> Weight {
         Weight::from_ref_time(17_443_346 as u64)
     }
-    fn execute_approved_call() -> Weight {
+    fn forward_approved_call() -> Weight {
         Weight::from_ref_time(17_443_346 as u64)
     }
 }
+
+pub trait ApprovedCallForwarder<AccountId, T> {
+    fn is_local() -> bool;
+    fn do_forward(
+        who: AccountId,
+        source_chain: String,
+        source_address: String,
+        contract_address: H160,
+        dest: MultiLocation,
+        call: Vec<u8>,
+    ) -> DispatchResult;
+}
+
+/// Local Forwarder Default Implementation
+pub struct ApprovedCallLocalForwarder;
```

Comment:
Let's make this clearer to distinct from the `Remote` one

```suggestion
pub struct LocalCallForwarder;
```